sql/grant_tables/user.go

```diff
@@ -0,0 +1,254 @@
+// Copyright 2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grant_tables
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/expression"
+	"github.com/dolthub/go-mysql-server/sql/in_mem_table"
+
+	"github.com/dolthub/vitess/go/sqltypes"
+)
+
+var (
+	userPkCols      = []uint16{0, 1}
+	errUserPkAssign = fmt.Errorf("the primary key for the `user` table expects a host and user string")
+)
+
+// UserPrimaryKey is a key that represents the primary key for the "user" Grant Table.
+type UserPrimaryKey struct {
+	Host string
+	User string
+}
+
+// AssignValues implements the interface in_mem_table.InMemTableDataKey.
+func (u UserPrimaryKey) AssignValues(vals ...interface{}) (in_mem_table.InMemTableDataKey, error) {
+	if len(vals) != 2 {
+		return u, errUserPkAssign
+	}
+	host, ok := vals[0].(string)
+	if !ok {
+		return u, errUserPkAssign
+	}
+	user, ok := vals[1].(string)
+	if !ok {
+		return u, errUserPkAssign
+	}
+	return UserPrimaryKey{
+		Host: host,
+		User: user,
+	}, nil
+}
+
+// RepresentedColumns implements the interface in_mem_table.InMemTableDataKey.
+func (u UserPrimaryKey) RepresentedColumns() []uint16 {
+	return userPkCols
+}
+
+var _ in_mem_table.InMemTableDataKey = UserPrimaryKey{}
+
+// createUserTable returns a new Grant Table representing the "user" table.
+func createUserTable() *grantTable {
```

Comment:
Probably makes sense to define everything in here as a package level var (unless this function is going to eventually take arguments?)

Reply:
Not planning on it taking arguments anytime soon. I don't quite want to pollute the namespace for every type and column that will be used (as I've got like 8 more tables to do), but I'll move a lot of this to an `init` function and store some of the results in package vars, which should be a happy middle ground.